packages/editor/src/client/atoms/comment/commentHistory.tsx

```diff
@@ -0,0 +1,109 @@
+import PlusIcon from '@rsuite/icons/legacy/Plus'
+import React, {useEffect, useState} from 'react'
+import {useTranslation} from 'react-i18next'
+import {FlexboxGrid} from 'rsuite'
+
+import {
+  CommentItemType,
+  CommentSort,
+  CommentState,
+  FullCommentFragment,
+  useCommentListQuery
+} from '../../api'
+import {CommentPreview} from './commentPreview'
+import {CreateCommentBtn} from './createCommentBtn'
+
+interface ChildCommentsProps {
+  comments?: FullCommentFragment[]
+  comment: FullCommentFragment
+  originCommentId?: string
+}
+
+function ChildComments({comments, comment, originCommentId}: ChildCommentsProps) {
+  if (!comments) {
+    return <></>
+  }
+  const childComments = comments.filter(tmpComment => tmpComment.parentComment?.id === comment.id)
+  return (
+    <div style={{marginTop: '20px', borderLeft: '1px lightgrey solid', paddingLeft: '20px'}}>
+      {childComments.map(childComment => (
+        <div key={childComment.id}>
+          <CommentPreview comment={childComment} expanded={childComment.id === originCommentId} />
+          {/* some fancy recursion */}
+          <ChildComments
+            comments={comments}
+            comment={childComment}
+            originCommentId={originCommentId}
+          />
+        </div>
+      ))}
+    </div>
+  )
+}
+
+interface CommentHistoryProps {
+  commentItemType: CommentItemType
+  commentItemID: string
+  commentId?: string
+}
+
+export function CommentHistory({commentId, commentItemType, commentItemID}: CommentHistoryProps) {
+  const {t} = useTranslation()
+  const [comments, setComments] = useState<FullCommentFragment[] | undefined>()
+  const {data, refetch} = useCommentListQuery({
```

Comment:
I guess this should use a LazyQuery as the useEffect below is called immediately